libcloud/compute/drivers/vcloud.py

```diff
@@ -1911,15 +2088,19 @@ def _add_vm_disk(self, vapp_or_vm_id, vm_disk):
             )
             self._wait_for_task_completion(res.object.get('href'))
 
-    def _change_vm_script(self, vapp_or_vm_id, vm_script):
-        if vm_script is None:
+    def _change_vm_script(self, vapp_or_vm_id, vm_script, vm_script_text=None):
+        if vm_script is None and vm_script_text is None:
             return
 
+        if vm_script_text is not None:
+            script = vm_script_text
+        else:
+            try:
+                script = open(vm_script).read()
```

Comment:
I know this piece of code was already here before, but it would be good to fix it to avoid leaking an open file handle.

So using ``with`` context manager.

```python
with open(vm_script, 'r') as fp:
    fp = content.read()
....
```

This will ensure file handle is always closed.

Reply:
Fixed